app/models/subject.rb

```diff
@@ -71,4 +71,23 @@ def retired_for_workflow?(workflow)
       false
     end
   end
+
+  def surrounding(subject_set_id, window=5, gap=1)
+    all_smses = SetMemberSubject.where(subject_set_id: subject_set_id).order(:priority)
```

Comment:
Try to avoid this `order(:priority)` as it'll order the whole set before filtering. Ideally we have a new index that covers subject_set_id and priority on SMS model i.e. `sms_subject_set_id_and_priority` and we know the current `priority` value and we can then walk forward / backward from this index location to get the gap cover without searching through the whole space. 

SO i think allow the current priority value as a param and change the current index `sms_subject_set_id` to be compound with `sms_subject_set_id_and_priority`. That should allow a faster index based find of the surrounding / adjacent priority sms records.

Reply:
Yeah, I said as much in the PR desc that the new/updated index is a given. Also mentioned that crafting this query in such a way as to return only the total window of interest is necessary to avoid loading and sorting whole sets.